cjs/src/api/controllers/demo.controller.js

```diff
@@ -0,0 +1,204 @@
+        )
+      );
+    })
+    .catch((err) => {
+      res.status(500).json({
+        error: `${err}`,
+      });
+      console.log(chalk.redBright(`\nError retriving ${item}s: ${err}\n`));
+    });
+};
+
+const createDemoItemController = (req, res) => {
+  createDemoItemService()
+    .then((doc) => {
+      res.status(201).json({
+        message: `${item} created successfully!`,
+        newItem: {
+          _id: doc._id,
+          name: doc.name,
+          age: doc.age,
+          request: {
+            type: "GET",
+            url: `http://localhost:3000/${routeName}/${doc._id}`,
+          },
+        },
+      });
+      console.log(
+        chalk.greenBright(
+          `\n${item} CREATED successfully! \n\nCreated ${item} url: http://localhost:3000/${routeName}/${doc._id}\n`
+        )
+      );
+    })
+    .catch((err) => {
+      res.status(500).json({
+        error: `${err}`,
+      });
+      console.log(chalk.redBright(`\nError saving ${item}: ${err}\n`));
+    });
+};
+
+const getOneDemoItemController = (req, res, next) => {
+  const id = req.params.demoId;
+  getOneDemoItemService(id)
+    .then((doc) => {
+      if (doc) {
+        res.status(200).json({
+          _id: doc._id,
+          name: doc.name,
+          age: doc.age,
+          request: {
+            type: "GET",
+            description: `Url link to all ${item}s`,
+            url: `http://localhost:3000/${routeName}/`,
+          },
+        });
+        console.log(
+          chalk.greenBright(
+            `\nGET request successful! \n\n${item} url: http://localhost:3000/${routeName}/${doc._id}\n`
+          )
+        );
+      } else {
+        console.log(chalk.redBright("\nNo record found for provided ID\n"));
+        return res.status(404).json({
+          message: "No record found for provided ID",
+        });
+      }
+    })
+    .catch((err) => {
+      res.status(500).json({
+        message: "Invalid ID",
+        error: `${err}`,
+      });
+      console.log(chalk.redBright(`\nError retriving ${item}: ${err}\n`));
+    });
+};
+
+const deleteDemoItemController = (req, res, next) => {
+  const id = req.params.demoId;
+  deleteDemoItemService(id)
+    .then((doc) => {
+      console.log(chalk.greenBright(`\n${item} DELETED successfully!\n`));
+      res.status(200).json({
+        message: `${item} deleted successfully!`,
+        request: {
+          type: "POST",
+          description: "Url link to make post request to",
+          url: `http://localhost:3000/${item}/`,
+          body: {
+            name: "String",
+            age: "Number",
+          },
+        },
+      });
+    })
+    .catch((err) => {
+      res.status(500).json({
+        message: `Error deleting ${item}`,
+        error: `${err}`,
+      });
+      console.log(chalk.redBright(`\nError deleting ${item}: ${err}\n`));
+    });
+};
+
+const partialUpdateDemoItemController = (req, res, next) => {
```

Comment:
- Same thing for the cjs template. Change the function names accordingly and rearrange as I have said in Review comments 1, 2 & 3 in #4